### PR TITLE
[FrameworkBundle] Finetune `AboutCommand`

### PR DESCRIPTION
**Pull request by JoppeDC**
| Q             | A
| ------------- | ---
| Branch?       | 7.2 
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| License       | MIT

- Make the AboutCommand show more details about the XDebug extension (Installed vs Enabled).
- Show the same info about the OPcache and APCu extensions

### CODE REVIEW

**Review by javiereguiluz — APPROVED** [review]
Good idea Joppe!

Just displaying `true`/`false` is not clear enough to distinct between installed or not and enabled or not. Thanks!

File: src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php [hunk 1]
```diff
@@ -81,9 +82,9 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             ['Architecture', (\PHP_INT_SIZE * 8).' bits'],
             ['Intl locale', class_exists(\Locale::class, false) && \Locale::getDefault() ? \Locale::getDefault() : 'n/a'],
             ['Timezone', date_default_timezone_get().' (<comment>'.(new \DateTimeImmutable())->format(\DateTimeInterface::W3C).'</>)'],
-            ['OPcache', \extension_loaded('Zend OPcache') && filter_var(\ini_get('opcache.enable'), \FILTER_VALIDATE_BOOL) ? 'true' : 'false'],
-            ['APCu', \extension_loaded('apcu') && filter_var(\ini_get('apc.enabled'), \FILTER_VALIDATE_BOOL) ? 'true' : 'false'],
-            ['Xdebug', \extension_loaded('xdebug') ? 'true' : 'false'],
+            ['OPcache', extension_loaded('Zend OPcache') ? (filter_var(ini_get('opcache.enable'), FILTER_VALIDATE_BOOLEAN) ? 'Enabled' : 'Not enabled') : 'Not installed'],
+            ['APCu', extension_loaded('apcu') ? (filter_var(ini_get('apc.enabled'), FILTER_VALIDATE_BOOLEAN) ? 'Enabled' : 'Not enabled') : 'Not installed'],
+            ['Xdebug', extension_loaded('xdebug') ? (ini_get('XDEBUG_MODE') ? 'Enabled (mode: ' . ini_get('XDEBUG_MODE') . ')' : 'Not enabled') : 'Not installed'],
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php]: `$mode = getenv('XDEBUG_MODE') ?: ini_get('xdebug.mode')`

when the mode is `off` xdebug is not enabled

**Review by ro0NL — COMMENTED** [review]
webprofiler should get a similar patch https://github.com/symfony/symfony/blob/a980a462f5eb1593d45f375d87b4bf06d6682666/src/Symfony/Component/HttpKernel/DataCollector/ConfigDataCollector.php#L57-L59

File: src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php [hunk 2]
```diff
@@ -81,11 +84,13 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
-            ['OPcache', \extension_loaded('Zend OPcache') && filter_var(\ini_get('opcache.enable'), \FILTER_VALIDATE_BOOL) ? 'true' : 'false'],
-            ['APCu', \extension_loaded('apcu') && filter_var(\ini_get('apc.enabled'), \FILTER_VALIDATE_BOOL) ? 'true' : 'false'],
-            ['Xdebug', \extension_loaded('xdebug') ? 'true' : 'false'],
+            ['OPcache', extension_loaded('Zend OPcache') ? (filter_var(ini_get('opcache.enable'), FILTER_VALIDATE_BOOLEAN) ? 'Enabled' : 'Not enabled') : 'Not installed'],
+            ['APCu', extension_loaded('apcu') ? (filter_var(ini_get('apc.enabled'), FILTER_VALIDATE_BOOLEAN) ? 'Enabled' : 'Not enabled') : 'Not installed'],
+            ['Xdebug', extension_loaded('xdebug') ? ($xdebugMode && $xdebugMode !== 'off' ? 'Enabled (' . $xdebugMode . ')' : 'Not enabled') : 'Not installed'],
         ];
 
+        
+
```
Comment (OskarStark) [inline src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php]:
```suggestion
```

**Review by xabbuh — APPROVED** [review]

**Review by ro0NL — CHANGES_REQUESTED** [review]
the same is in webprofiler, it should be in sync

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
> the same is in webprofiler, it should be in sync

After this comment 👍

**Comment by OskarStark** [comment]
@JoppeDC open for finishing this PR? Thanks

**Comment by JoppeDC** [comment]
> @JoppeDC open for finishing this PR? Thanks

Absolutely. I'll have a look at the profiler changes

File: src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php [hunk 3]
```diff
@@ -81,9 +84,9 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             ['Architecture', (\PHP_INT_SIZE * 8).' bits'],
             ['Intl locale', class_exists(\Locale::class, false) && \Locale::getDefault() ? \Locale::getDefault() : 'n/a'],
             ['Timezone', date_default_timezone_get().' (<comment>'.(new \DateTimeImmutable())->format(\DateTimeInterface::W3C).'</>)'],
-            ['OPcache', \extension_loaded('Zend OPcache') && filter_var(\ini_get('opcache.enable'), \FILTER_VALIDATE_BOOL) ? 'true' : 'false'],
-            ['APCu', \extension_loaded('apcu') && filter_var(\ini_get('apc.enabled'), \FILTER_VALIDATE_BOOL) ? 'true' : 'false'],
-            ['Xdebug', \extension_loaded('xdebug') ? 'true' : 'false'],
+            ['OPcache', extension_loaded('Zend OPcache') ? (filter_var(ini_get('opcache.enable'), FILTER_VALIDATE_BOOLEAN) ? 'Enabled' : 'Not enabled') : 'Not installed'],
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php]: please keep the `\` in place

File: src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php [hunk 4]
```diff
@@ -57,6 +58,8 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $buildDir = $kernel->getCacheDir();
         }
 
+        $xdebugMode = getenv('XDEBUG_MODE') ?: ini_get('xdebug.mode');
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php]:
```suggestion
        $xdebugMode = getenv('XDEBUG_MODE') ?: \ini_get('xdebug.mode');
```

File: src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php [hunk 5]
```diff
@@ -81,9 +84,9 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             ['Architecture', (\PHP_INT_SIZE * 8).' bits'],
             ['Intl locale', class_exists(\Locale::class, false) && \Locale::getDefault() ? \Locale::getDefault() : 'n/a'],
             ['Timezone', date_default_timezone_get().' (<comment>'.(new \DateTimeImmutable())->format(\DateTimeInterface::W3C).'</>)'],
-            ['OPcache', \extension_loaded('Zend OPcache') && filter_var(\ini_get('opcache.enable'), \FILTER_VALIDATE_BOOL) ? 'true' : 'false'],
-            ['APCu', \extension_loaded('apcu') && filter_var(\ini_get('apc.enabled'), \FILTER_VALIDATE_BOOL) ? 'true' : 'false'],
-            ['Xdebug', \extension_loaded('xdebug') ? 'true' : 'false'],
+            ['OPcache', extension_loaded('Zend OPcache') ? (filter_var(ini_get('opcache.enable'), FILTER_VALIDATE_BOOLEAN) ? 'Enabled' : 'Not enabled') : 'Not installed'],
+            ['APCu', extension_loaded('apcu') ? (filter_var(ini_get('apc.enabled'), FILTER_VALIDATE_BOOLEAN) ? 'Enabled' : 'Not enabled') : 'Not installed'],
+            ['Xdebug', extension_loaded('xdebug') ? ($xdebugMode && $xdebugMode !== 'off' ? 'Enabled (' . $xdebugMode . ')' : 'Not enabled') : 'Not installed'],
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php]:
```suggestion
            ['Xdebug', extension_loaded('xdebug') ? ($xdebugMode && $xdebugMode !== 'off' ? 'Enabled ('.$xdebugMode.')' : 'Not enabled') : 'Not installed'],
```

**Review by nicolas-grekas — COMMENTED** [review]
Status: needs work

**Comment by javiereguiluz** [comment]
@JoppeDC if you have some time, we need some last effort from you to fix some minor issues here. Then, we'll merge this in time for Symfony 7.2 release. Thanks!

**Comment by JoppeDC** [comment]
@javiereguiluz I've quickly fixed the feedback issues that were still open.
I have not had the time (and also i totally forgot) to check out the implementation in the Profiler tho

**Review by GromNaN — APPROVED** [review]
LGTM

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/config.html.twig [hunk 6]
```diff
@@ -250,17 +250,17 @@
     <div class="metrics">
         <div class="metric-group">
             <div class="metric">
-                <span class="value value-is-icon {{ not collector.haszendopcache ? 'value-shows-no-color' }}">{{ source('@WebProfiler/Icon/' ~ (collector.haszendopcache ? 'yes' : 'no') ~ '.svg') }}</span>
+                <span class="value value-is-icon {{ not collector.haszendopcache ? 'value-shows-no-color' }}" title="{{ collector.zendopcachestatus | default('') }}">{{ source('@WebProfiler/Icon/' ~ (collector.haszendopcache ? 'yes' : 'no') ~ '.svg') }}</span>
```
Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/config.html.twig]:
```suggestion
                <span class="value value-is-icon {{ not collector.haszendopcache ? 'value-shows-no-color' }}" title="{{ collector.zendopcachestatus|default('') }}">{{ source('@WebProfiler/Icon/' ~ (collector.haszendopcache ? 'yes' : 'no') ~ '.svg') }}</span>
```

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/config.html.twig [hunk 7]
```diff
@@ -250,17 +250,17 @@
...
                 <span class="label">OPcache</span>
             </div>
 
             <div class="metric">
-                <span class="value value-is-icon {{ not collector.hasapcu ? 'value-shows-no-color' }}">{{ source('@WebProfiler/Icon/' ~ (collector.hasapcu ? 'yes' : 'no') ~ '.svg') }}</span>
+                <span class="value value-is-icon {{ not collector.hasapcu ? 'value-shows-no-color' }}" title="{{ collector.apcustatus | default('') }}">{{ source('@WebProfiler/Icon/' ~ (collector.hasapcu ? 'yes' : 'no') ~ '.svg') }}</span>
```
Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/config.html.twig]:
```suggestion
                <span class="value value-is-icon {{ not collector.hasapcu ? 'value-shows-no-color' }}" title="{{ collector.apcustatus|default('') }}">{{ source('@WebProfiler/Icon/' ~ (collector.hasapcu ? 'yes' : 'no') ~ '.svg') }}</span>
```

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/config.html.twig [hunk 8]
```diff
@@ -250,17 +250,17 @@
...
                 <span class="label">APCu</span>
             </div>
 
             <div class="metric">
-                <span class="value value-is-icon {{ not collector.hasxdebug ? 'value-shows-no-color' }}">{{ source('@WebProfiler/Icon/' ~ (collector.hasxdebug ? 'yes' : 'no') ~ '.svg') }}</span>
+                <span class="value value-is-icon {{ not collector.hasxdebug ? 'value-shows-no-color' }}" title="{{ collector.xdebugstatus | default('') }}">{{ source('@WebProfiler/Icon/' ~ (collector.hasxdebug ? 'yes' : 'no') ~ '.svg') }}</span>
```
Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/config.html.twig]:
```suggestion
                <span class="value value-is-icon {{ not collector.hasxdebug ? 'value-shows-no-color' }}" title="{{ collector.xdebugstatus|default('') }}">{{ source('@WebProfiler/Icon/' ~ (collector.hasxdebug ? 'yes' : 'no') ~ '.svg') }}</span>
```

**Review by fabpot — APPROVED** [review]
Small CS fixes needed.

**Comment by fabpot** [comment]
Thank you @JoppeDC.

**Pull request merged**